src/main/scala/uk/pkerrigan/darwin/Message.scala

```diff
@@ -0,0 +1,63 @@
+package uk.pkerrigan.darwin
+
+import scala.xml.{Node, XML}
+
+sealed trait Time
+case class EstimatedTime(time: String) extends Time
+case class ActualTime(time: String) extends Time
+
+sealed trait Location {
+  def tiploc: String
+}
+case class Station(tiploc: String, arrivalTime: Time, departureTime: Time) extends Location
+case class Waypoint(tiploc: String, passTime: Time) extends Location
+
+sealed trait Message {
+  def origin: String
+}
+case class ScheduleUpdate(origin: String, runId: String, locations: List[Location]) extends Message
+
+object Message {
+  def parseMessage(message: String): Option[Message] =
```

Comment:
Rather than returning `Option`s throughout you could call `.toRight` in various places and return `Either[ParseError, A]` where `ParseError` is a case class or string or something